### ISSUE TITLE
Volume alignment

### ISSUE THREAD

**Issue by janden** [ComputationalCryoEM/ASPIRE-Python#917]
As a followup to #901, we should start thinking about image alignment. The goal should have something that is functional but maybe slow that we can optimize later on.

For some inspiration EMAN2 has a C++ aligner that they use which is pretty fast: https://github.com/cryoem/eman2/blob/ce7a9f0540c10bdaee859383dee73dfcb5ce5612/libEM/aligner.cpp, but this is beyond the level of complexity that we want. There's also a short discussion of the problem at the end of the FTK paper: https://arxiv.org/pdf/1905.12317.pdf.  Finally, there's the new approach proposed by Yoel: https://www.cambridge.org/core/services/aop-cambridge-core/content/view/3F7560D96BD6FD4900053957ADAB61AC/S2633903X23000089a.pdf/three-dimensional-alignment-of-density-maps-in-cryo-electron-microscopy.pdf.

Simplest thing to start with would be to simply rotate one of the volumes over a uniform grid of Euler angles and maximize the correlation. This will give us a (slow) baseline that we can improve upon successively.

**Comment by garrettwrong** [comment]
Linking #945 .  I think it still needs to be determined how well this works in practice for non identical volumes etc, but seems promising.

**Comment by janden** [comment]
Agreed. Would be good to have a (slow) reference aligner as well to compare to.
